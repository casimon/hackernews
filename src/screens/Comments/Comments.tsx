import React from 'react'
import { useSelector } from 'react-redux'

import './Comments.css'
import Comment from '../../components/Comment/Comment'
import Header from '../../components/Header/Header'

import { useFetching } from '../../hooks/useFetching'
import { getTopComments } from '../../reducers'

interface Comment {
  by: string
  id: number
  kids: Array<number>
  parent: number
  text: string
  time: number
  type: string
}

interface TopComments {
  topComments: Array<Comment>
}

function Comments() {
  useFetching(getTopComments)
  const topComments = useSelector((state: TopComments) => state.topComments)
  return (
    <div className='Page'>
      <Header />
      <article className='Page-body'>
        <section>
          <h2>Top Comments</h2>
          {topComments.length > 0 ? (
            topComments.map((item: Comment, index: number) => (
              <Comment text={item.text} by={item.by} key={index} />
            ))
          ) : (
            <p>Loading...</p>
          )}
        </section>
      </article>
    </div>
  )
}

export default Comments
