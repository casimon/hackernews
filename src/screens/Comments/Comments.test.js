import React from 'react'

import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import Comments from './Comments'

const mockStore = configureStore([])

jest.mock('../../hooks/useFetching')

const init = (store) =>
  render(
    <Provider store={store}>
      <Comments />
    </Provider>
  )

describe('My connected React-Redux component', () => {
  test('should render comments page with default state from Redux store', () => {
    const store = mockStore({
      topComments: [],
      errors: [],
    })

    const { container } = init(store)
    expect(container).toMatchSnapshot()
  })

  test('should render comments page with top comments', () => {
    const store = mockStore({
      topComments: [
        {
          by: 'test1',
          id: 1,
          kids: [1],
          parent: 1,
          time: 1,
          text: 'Test 1',
          type: 'story',
        },
      ],
      errors: [],
    })
    const { container } = init(store)
    expect(container).toMatchSnapshot()
  })
})
