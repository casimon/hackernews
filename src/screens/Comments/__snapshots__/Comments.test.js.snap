// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My connected React-Redux component should render comments page with default state from Redux store 1`] = `
<div>
  <div
    class="Page"
  >
    <header
      class="Header"
    >
      <img
        alt="Logo"
        src="Y_Combinator_logo.svg"
      />
      <h1>
        Hacker News
      </h1>
      <div
        class="Header-nav"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/comments"
        >
          Comments
        </a>
      </div>
    </header>
    <article
      class="Page-body"
    >
      <section>
        <h2>
          Top Comments
        </h2>
        <p>
          Loading...
        </p>
      </section>
    </article>
  </div>
</div>
`;

exports[`My connected React-Redux component should render comments page with top comments 1`] = `
<div>
  <div
    class="Page"
  >
    <header
      class="Header"
    >
      <img
        alt="Logo"
        src="Y_Combinator_logo.svg"
      />
      <h1>
        Hacker News
      </h1>
      <div
        class="Header-nav"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/comments"
        >
          Comments
        </a>
      </div>
    </header>
    <article
      class="Page-body"
    >
      <section>
        <h2>
          Top Comments
        </h2>
        <div
          class="Comment"
        >
          <div>
            Test 1
          </div>
          <p>
            By 
            test1
          </p>
        </div>
      </section>
    </article>
  </div>
</div>
`;
